src/main/java/com/deque/axe/AXE.java

```diff
@@ -355,18 +355,31 @@ private String getAxeSnippet(String context, String options) {
 		 * @param context A WebElement to test
 		 * @return An aXe results document
 		 */
-		public JSONObject analyze(final WebElement context) {
+		public JSONObject analyze(final WebElement... context) {
 			String snippet = getAxeSnippet("arguments[0]", options);
-			return execute(snippet, context);
+			return execute(snippet, (Object) context);
 		}
 
 		private JSONObject execute(final String command, final Object... args) {
 			AXE.inject(this.driver, this.script, this.skipFrames);
 
 			this.driver.manage().timeouts().setScriptTimeout(this.timeout, TimeUnit.SECONDS);
 
-			Object response = ((JavascriptExecutor) this.driver).executeAsyncScript(command, args);
-			JSONObject result = new JSONObject((Map) response);
+			JSONObject result= new JSONObject();
+
+			if (args.length > 0)
+			    for (Object arg: args) {
```

Comment:
1. Any way around not having to execute axe for each webelement ? Concerned about the perf implications here - since axe would be injected into the page for each time etc. 

2. Would it be possible to convert the webelements into CSS selectors and use it along with the `include` option ?

Reply:
Good questions!
As far as I know, you can't pull the CSS selector from WebElement. Most likely, can write a js script to get the path to the element. I don't think it's easy, otherwise it would have existed long ago) because it would have solved a lot of interesting problems when testing the UI.

But I can actually pass the entire list of elements at once. Whether there will be an increase in performance is difficult to say. But it will definitely look better and easier. So I change code.